src/hooks/hookResources.ts

```diff
@@ -63,42 +86,69 @@ export function getFhirResource(token: string, req: TypedRequestBody) {
   });
 }
 
-
+const createErrorCard = (summary: string) => {
+  return {
+    cards: [
+      {
+        summary: summary, 
+        indicator: 'warning',
+        source: {
+          label: 'REMS Intermediary',
+          url: 'http://localhost:3003'
+        }
+      }
+    ]
+  }
+}
 // handles all hooks, any supported hook should pass through this function
 export async function handleHook(
   req: TypedRequestBody,
   res: TypedResponseBody,
   hookPrefetch: ServicePrefetch,
   contextRequest: FhirResource | undefined
 ) {
-  const url = config?.general?.remsAdminHookPath + req?.body?.hook;
-  console.log('rems-admin hook url: ' + url);
-
-  //TODO: lookup hook url based on medication from the database
-  //  look at rems-admin hookResources code to determine how to get the medication code
-
-  const forwardData = (hook: Hook) => {
-    // remove the auth token before any forwarding occurs
-    delete hook.fhirAuthorization;
-    const options = {
-      method: 'POST',
-      data: hook
-    };
-    const response = axios(url, options);
-    response.then(e => {
-      res.json(e.data);
-    });
-  }
+  if(contextRequest && contextRequest.resourceType === 'MedicationRequest'){
+    const drugCode = getDrugCodeFromMedicationRequest(contextRequest);
 
-  let hook: Hook = req.body;
-  if(hook.fhirAuthorization && hook.fhirServer && hook.fhirAuthorization.access_token) {
-    hydrate(getFhirResource, hookPrefetch, hook).then((hydratedPrefetch) => {
-      if(hydratedPrefetch) {
-        hook.prefetch = hydratedPrefetch;
+    const forwardData = (hook: Hook, url: string) => {
+      // remove the auth token before any forwarding occurs
+      delete hook.fhirAuthorization;
+      const options = {
+        method: 'POST',
+        data: hook
+      };
+      const response = axios(url, options);
+      response.then(e => {
+        res.json(e.data);
+      });
+    }
+    if(drugCode) {
+      let hook: Hook = req.body;
+      const serviceUrl = getServiceUrl(drugCode, hook.fhirServer?.toString());
+      if(serviceUrl) {
+        const url = serviceUrl + hook.hook;
+        console.log('rems-admin hook url: ' + url);
+        if(hook.fhirAuthorization && hook.fhirServer && hook.fhirAuthorization.access_token) {
+          hydrate(getFhirResource, hookPrefetch, hook).then((hydratedPrefetch) => {
+            if(hydratedPrefetch) {
+              hook.prefetch = hydratedPrefetch;
+            }
+            forwardData(hook, url);
+          })
+        } else {
+          forwardData(hook, url);
+        }
+      } else {
+        // unsupported drug code, TODO - what to do when we don't have a service url
+        console.log("niope");
```

Comment:
Perhaps a better log here would be good, like the 'Unsupported Drug Code' from below with the code.